tools/update_tool

```diff
@@ -0,0 +1,466 @@
+    Can apply (and will by default) filters to the image-info to get rid of some
+    volatile UUIds that aren't important.
+    """
+    manifest = None
+    imi = None
+    if not db_entry:
+        return None, None
+    if db_entry["image-info"]:
+        if do_filter:
+            imi = filter_fn(db_entry["image-info"])
+        else:
+            imi = db_entry["image-info"]
+    if db_entry["manifest"]:
+        manifest = db_entry["manifest"]
+    return manifest, imi
+
+
+def get_symbol(label):
+    """
+    Get an icon to sum up what changed
+    """
+    if label == "added":
+        return "🆕"
+    if label == "updated":
+        return "🔁"
+    if label == "removed":
+        return "🗑️"
+    if label == "unchanged":
+        return "💾"
+    return None
+
+
+
+
+def progress(count, total, prefix=''):
+    """
+    Prints a nice progress bar that takes up to the all width of the terminal
+    """
+    bar_len = os.get_terminal_size().columns - 10 - len(prefix)
+    filled_len = int(round(bar_len * count / float(total)))
+
+    percents = round(100.0 * count / float(total), 1)
+    barr = '-' * (filled_len-1) + ">" + ' ' * (bar_len - filled_len)
+
+    sys.stdout.write(f'{prefix}{percents}%[{barr}]\r')
+    sys.stdout.flush()
+
+
+def erase_bar():
+    """
+    Erases the progress bar
+    """
+    sys.stdout.write(f'{" "*os.get_terminal_size().columns}\r')
+    sys.stdout.flush()
+
+
+def ls_command(args):
+    """
+    Lists all the changed files between this and the previous commit and puts
+    some icons for the user to know what changed and what not
+    """
+    files = []
+    lst = list_changed_files()
+    for i, file in enumerate(lst):
+        filename = file.split("/")[-1].split(".")[0]
+        prev = get_db_entry_at_previous_version(file)
+        curr = get_db_entry_at_current_version(file)
+
+        if not (prev or curr):
+            continue
+
+        prev_manifest, prev_imi = extract(prev, do_filter=not args.no_filter)
+        curr_manifest, curr_imi = extract(curr, do_filter=not args.no_filter)
+
+        manifest_status = compare(prev_manifest, curr_manifest)
+        image_info_status = compare(prev_imi, curr_imi)
+
+        if manifest_status != "unchanged" or image_info_status != "unchanged":
+            ms = get_symbol(compare(prev_manifest, curr_manifest))
+            ims = get_symbol(compare(prev_imi, curr_imi))
+            files.append(f"{ms}{ims} {filename}")
+        progress(i+1, len(lst), "listing changed files: ")
+    erase_bar()
+    print("(manifest)(image-info) file:")
+    print("🆕: added 🔁: updated 🗑️: removed 💾: unchanged\n")
+
+    if args.l:
+        for f in files:
+            print(f)
+    else:
+        print(columnify.columnify(
+            items=files,
+            line_width=os.get_terminal_size().columns))
+
+
+def diff_command(args):
+    """
+    Diffs either some files or all the files with the difftool provided by the
+    user. Default tool is vimdiff
```

Comment:
A small nitpick: `diff_command` doesn't know anything about the default diff tool, that's set higher.